src/parse/parser.rs

```diff
@@ -481,16 +471,21 @@ impl<'help, 'app> Parser<'help, 'app> {
                 }
             }
 
-            let positional_count = self
-                .app
-                .args
-                .keys
-                .iter()
-                .filter(|x| x.key.is_position())
-                .count();
-            let is_second_to_last = positional_count > 1 && (pos_counter == (positional_count - 1));
```

Comment:
pos_counter begins with 1. So this can be changed to `pos_counter + 1 == positional_count`